Backpack-SwiftUI/Calendar/Classes/CalendarSignleSelectionState.swift

```diff
@@ -0,0 +1,28 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+/// The `CalendarSingleSelectionState` enum represents the different states of a single selection.
+public enum CalendarSingleSelectionState {
+    /// The state, with single date selection
+    case single(Date?)
```

Comment:
I think this should probably not be optional right? the optional would be on the enum itself, but not the date.
reasoning being:
if there's a single date selected, it makes no sense that that seleciton has no date

Reply:
In both Calendar selection types (range/single) there's a use-case for not having a date being selected (when dates being cleared). So this optional date is still valid.

|      Single      |     Range    |
|---------------------|------------------|
|       ![Simulator Screenshot - iPhone SE (3rd generation) - 2024-11-26 at 10 38 16](https://github.com/user-attachments/assets/a3a45f34-21f0-4fca-aa16-ac8e1db7d844)         |     ![Simulator Screenshot - iPhone SE (3rd generation) - 2024-11-26 at 10 38 10](https://github.com/user-attachments/assets/17ff60ef-3546-407b-a8f9-bbd50a0c5eb5)          |

Reply:
Cool. Right. I applied the change. ✅